### PR TITLE
Added Manip Deployment Potentiometer

### PR DESCRIPTION
**Pull request by 1000DollarBill**
Added potentiometer to deployment of manipulator commands and the manipulator subsystem. Deleted unneeded method, and changed manipulator into a PIDSubsysytem.

### CODE REVIEW

File: src/main/java/frc/team166/robot/RobotMap.java [hunk 1]
```diff
@@ -74,12 +76,19 @@
         public static final String K_I = "kI";
         public static final String K_D = "kD";
         public static final String K_F = "kF";
+        public static final String K_PATRICK = "kPatrick";
```
Comment (ghost) [inline src/main/java/frc/team166/robot/RobotMap.java]: Let's name this K_P_Manipulator... And apply that the the others ones too...

File: src/main/java/frc/team166/robot/subsystems/Manipulator.java [hunk 2]
```diff
@@ -188,4 +213,36 @@ protected void interrupted() {
...
+
+            @Override
+            protected void execute() {
+
+            }
+
+            @Override
+            protected boolean isFinished() {
+
+                return isTimedOut();
```
Comment (ghost) [inline src/main/java/frc/team166/robot/subsystems/Manipulator.java]: I think you need to set this somewhere first? What is the timeout value??

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/main/java/frc/team166/robot/subsystems/Manipulator.java (hunk 2 above)
Comment (1000DollarBill) [reply src/main/java/frc/team166/robot/subsystems/Manipulator.java]: No, it is just a function. It isn't a variable or anything that needs to be declared somewhere else.

File: src/main/java/frc/team166/robot/RobotMap.java (hunk 1 above)
Comment (1000DollarBill) [reply src/main/java/frc/team166/robot/RobotMap.java]: Why do you have to ruin all of my fun?

File: src/main/java/frc/team166/robot/subsystems/Manipulator.java (hunk 2 above)
Comment (1000DollarBill) [reply src/main/java/frc/team166/robot/subsystems/Manipulator.java]: Nevermind I'm wrong, just fixed it.

File: src/main/java/frc/team166/robot/RobotMap.java [hunk 3]
```diff
@@ -37,6 +38,7 @@
         //changes input ports into integers
         public final static int tempestgyro = 1;
         public final static int IR = 2;
+        public final static int potentiometer = 3;
```
Comment (bot190) [inline src/main/java/frc/team166/robot/RobotMap.java]: This should be MANIPULATOR_POTENTIOMETER

File: src/main/java/frc/team166/robot/RobotMap.java [hunk 4]
```diff
@@ -38,6 +39,7 @@
         public final static int tempestgyro = 1;
         public final static int IR = 2;
         public final static int liftEncoder = 3;
+        public final static int MANIPULATOR_POTENTIOMETER = 3;
```
Comment (msoucy) [inline src/main/java/frc/team166/robot/RobotMap.java:42]: Lowercase, and change the value

File: src/main/java/frc/team166/robot/RobotMap.java [hunk 5]
```diff
@@ -78,16 +80,22 @@
         public static final String K_I = "kI";
         public static final String K_D = "kD";
         public static final String K_F = "kF";
-        public static final String UP_MAX_SPEED = "upMaxSpeed";
-        public static final String DOWN_MAX_SPEED = "downMaxSpeed";
+        public static final String K_P_MANIPULATOR = "kPatrick";
+        public static final String K_I_MANIPULATOR = "kIbsen";
+        public static final String K_D_MANIPULATOR = "kDerrick";
+        public static final String K_F_MANIPULATOR = "kForbath";
```
Comment (msoucy) [inline src/main/java/frc/team166/robot/RobotMap.java]: ... No.

File: src/main/java/frc/team166/robot/subsystems/Manipulator.java [hunk 6]
```diff
@@ -24,26 +26,37 @@
...
     double DIST_PER_PULSE_INTAKE = (((ROLLER_RADIUS * 2.0 * Math.PI) / 1024.0) / 12.0); //feet
     double OPTIMAL_MOTOR_RATE = 6.81; //ft/s
 
+    private static double kP_Manipulator = Preferences.getInstance().getDouble(RobotMap.Preferences.K_P_MANIPULATOR, 1);
+    private static double kI_Manipulator = Preferences.getInstance().getDouble(RobotMap.Preferences.K_I_MANIPULATOR, 1);
+    private static double kD_Manipulator = Preferences.getInstance().getDouble(RobotMap.Preferences.K_D_MANIPULATOR, 1);
+    private static double kF_Manipulator = Preferences.getInstance().getDouble(RobotMap.Preferences.K_F_MANIPULATOR, 1);
+
     public Manipulator() {
+        super("Manipulator (AKA Chadwick)", kP_Manipulator, kI_Manipulator, kD_Manipulator, kF_Manipulator);
```
Comment (msoucy) [inline src/main/java/frc/team166/robot/subsystems/Manipulator.java:56]: ..... I'll tolerate it.

**Review by msoucy — CHANGES_REQUESTED** [review]

**Review by msoucy — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**